code/ui/generalhud/mapselectionmenu/MapSelectionMenu.cs

```diff
@@ -0,0 +1,127 @@
+using System.Collections.Generic;
+
+using Sandbox;
+using Sandbox.UI;
+using Sandbox.UI.Construct;
+
+using TTTReborn.Globals;
+
+namespace TTTReborn.UI
+{
+    public class MapSelectionMenu : Panel
+    {
+        public static MapSelectionMenu Instance;
+
+        private readonly Panel _mapWrapper;
+        private readonly List<MapPanel> _mapPanels;
+
+        public MapSelectionMenu() : base()
+        {
+            Instance = this;
+
+            StyleSheet.Load("/ui/generalhud/mapselectionmenu/MapSelectionMenu.scss");
+
+            AddClass("text-shadow");
+
+            AddClass("background-color-secondary");
+            AddClass("opacity-heavy");
+            AddClass("fullscreen");
+
+            Add.TranslationLabel("VOTE_NEXT_MAP", "title");
+
+            _mapPanels = new();
+
+            _mapWrapper = new Panel(this);
+            _mapWrapper.AddClass("map-wrapper");
+
+            InitMapPanels();
+
+            Enabled = false;
+        }
+
+        [Event(Events.TTTEvent.MapSelectionHandler.MapImagesChange)]
+        private void OnMapImagesChange()
+        {
+            InitMapPanels();
+        }
+
+        private void InitMapPanels()
+        {
+            IDictionary<string, string> mapImages = Gamemode.Game.Instance.MapSelection.MapImages;
+            foreach (KeyValuePair<string, string> mapImage in mapImages)
+            {
+                if (_mapPanels.Exists((mapPanel) => mapPanel.MapName == mapImage.Key))
+                {
+                    continue;
+                }
+
+                MapPanel panel = new(mapImage.Key, mapImage.Value)
+                {
+                    Parent = _mapWrapper
+                };
+
+                _mapPanels.Add(panel);
+            }
+        }
+
+        public override void Tick()
+        {
+            base.Tick();
+
+            if (!Enabled)
+            {
+                return;
+            }
+
+            IDictionary<long, string> playerIdMapVote = Gamemode.Game.Instance.MapSelection.PlayerIdMapVote;
+
+            IDictionary<string, int> mapToVoteCount = Map.MapSelectionHandler.GetTotalVotesPerMap(playerIdMapVote);
+
+            bool hasLocalClientVoted = playerIdMapVote.ContainsKey(Local.Client.PlayerId);
+
+            for (int i = 0; i < _mapPanels.Count; ++i)
```

Comment:
Wouldn't a foreach loop work as well?

Reply:
good catch, yeah it was from a previous iteration where I needed the index for something.